src/claude_client.rs

```diff
@@ -0,0 +1,333 @@
+    current_user_message: &str,
+  ) -> Result<String> {
+    // If no Discord history, use the simple send_message method
+    if discord_messages.is_empty() {
+      return self
+        .send_message(conversation_key, current_user_message)
+        .await;
+    }
+
+    // Get or create conversation context and initialize with Discord history if new
+    {
+      let context = self.get_or_create_conversation(conversation_key);
+
+      // If this is a new conversation, initialize it with Discord history
+      if context.messages.is_empty() {
+        for (author_name, content) in discord_messages {
+          if author_name.starts_with("bot:") {
+            context.add_message("assistant", content);
+          } else {
+            context.add_message("user", &format!("User {} says: {}", author_name, content));
+          }
+        }
+      }
+
+      // Add the current message
+      context.add_message("user", current_user_message);
+    }
+
+    // Clone the context data for the Claude call
+    let context_data = {
+      let context = self.get_or_create_conversation(conversation_key);
+      context.clone()
+    };
+
+    let response = self.call_claude_cli(&context_data).await?;
+
+    // Add the assistant response
+    {
+      let context = self.get_or_create_conversation(conversation_key);
+      context.add_message("assistant", &response.content);
+    }
+
+    // Save the updated conversation to persistence
+    self.save_conversation_to_persistence(conversation_key);
+
+    Ok(response.content)
+  }
+
+  fn save_conversation_to_persistence(&self, conversation_key: &ConversationId) {
+    if let Some(context) = self.conversations.get(conversation_key) {
+      if let Err(e) = self
+        .persistence
+        .save_conversation(conversation_key, context)
+      {
+        error!(
+          "Failed to save conversation {} to persistence: {}",
+          conversation_key, e
+        );
+      }
+    }
+  }
+
+  async fn call_claude_cli(&self, context: &ConversationContext) -> Result<ClaudeResponse> {
+    let claude_path = self
+      .claude_path
+      .as_ref()
+      .ok_or_else(|| anyhow!("Claude CLI not available - chat mode is disabled"))?;
+
+    let mut conversation_text = String::new();
+
+    for message in &context.messages {
+      conversation_text.push_str(&format!("{}: {}\n\n", message.role, message.content));
+    }
+
+    let system_prompt = "You are a silly, chatty cat with sometimes snarky responses. Always respond in 3 sentences or less.";
+
+    let mut child = Command::new(claude_path)
+      .args([
+        "--system-prompt",
+        system_prompt,
+        "--print",
+        "--output-format",
+        "json",
+        &conversation_text,
+      ])
+      .stdin(Stdio::piped())
+      .stdout(Stdio::piped())
+      .stderr(Stdio::piped())
+      .spawn()
+      .map_err(|e| anyhow!("Failed to spawn Claude CLI process: {}", e))?;
```

Comment:
`debug!()` log the built command before spawning it so we can recreate any issues needed